test/index.test.js

```diff
@@ -1,9 +1,309 @@
+            try {
+                notifier = new EmailNotifier(configMock, serverMock, 'build_status_test');
+                assert.fail('should not get here');
+            } catch (err) {
+                assert.instanceOf(err, Error);
+                assert.equal(err.name, 'ValidationError');
+            }
+        });
+
+        it('validates port', () => {
+            configMock.port = 'nonIntegerPort';
+            try {
+                notifier = new EmailNotifier(configMock, serverMock, 'build_status_test');
+                assert.fail('should not get here');
+            } catch (err) {
+                assert.instanceOf(err, Error);
+                assert.equal(err.name, 'ValidationError');
+            }
+        });
+
+        it('validates the from email', () => {
+            configMock.from = 'nonEmailString';
+            try {
+                notifier = new EmailNotifier(configMock, serverMock, 'build_status_test');
+                assert.fail('should not get here');
+            } catch (err) {
+                assert.instanceOf(err, Error);
+                assert.equal(err.name, 'ValidationError');
+            }
+        });
+
+        it('validates config format', () => {
+            configMock = ['this', 'is', 'wrong'];
+
+            try {
+                notifier = new EmailNotifier(configMock, serverMock, 'build_status_test');
+                assert.fail('should not get here');
+            } catch (err) {
+                assert.instanceOf(err, Error);
+                assert.equal(err.name, 'ValidationError');
+            }
+        });
+    });
+
+    describe('buildData is validated', () => {
+        before(() => {
+            mockery.enable({
+                useCleanCache: true,
+                warnOnUnregistered: false
+            });
+
+            nodemailerMock = getNodemailerMock();
+            mockery.registerMock('nodemailer', nodemailerMock);
+        });
+
+        after(() => {
+            mockery.disable();
+        });
+
+        beforeEach(() => {
+            // eslint-disable-next-line global-require
+            const EmailNotifier = require('../index');
+
+            serverMock = new Hapi.Server();
+            configMock = {
+                host: 'testing.aserver.com',
+                port: 25,
+                from: 'user@email.com'
+            };
+            buildDataMock = {
+                settings: {
+                    addresses: ['notify.me@email.com', 'notify.you@email.com'],
+                    statuses: ['SUCCESS', 'FAILURE']
+                },
+                status: 'SUCCESS'
+            };
+
+            notifier = new EmailNotifier(configMock, serverMock, 'build_status_test');
+        });
+
+        afterEach(() => {
+            mockery.deregisterAll();
+            mockery.resetCache();
+        });
+
+        it('validates status', () => {
+            buildDataMock.status = 22;
+            serverMock.event('build_status_test');
+            try {
```

Comment:
Gotcha.

So looking at the example, the main difference is that it's constructing an object with the `new` keyword. Instantiating a new object is outside of a Promise chain. Promise chains have this nice feature where if an error is thrown in the chain, it will catch it and propagate it through the chain until something catches it or the script throws it.

In this case of `notifier.notify()`, you're in this Promise chain safety net. If an error is throw, you should be able to catch it afterwards. Like so:

```
notifier.notify()
.then(() => {
  assert.fail();
}, (err) => {
  // check the error here.
})

// An alternative
notifier.notify()
.catch((err) => {
  // check the error here
});
```